iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/TerminalRefXml.java

```diff
@@ -18,21 +20,52 @@
  */
 public final class TerminalRefXml {
 
+    /**
+     * @deprecated Use {@link #writeTerminalRef(Connectable, Terminal, NetworkXmlWriterContext, String)} instead
+     */
+    @Deprecated
     public static void writeTerminalRef(Terminal t, NetworkXmlWriterContext context, String elementName) throws XMLStreamException {
-        writeTerminalRef(t, context, context.getVersion().getNamespaceURI(), elementName);
+        writeTerminalRef(null, t, context, context.getVersion().getNamespaceURI(), elementName);
```

Comment:
I don't know if passing null is OK, that could throw an exception? Is there a unit test to ensure that is OK?

Reply:
Yes, re-reading the code I think this produces a nullpointerexception in the comparison with the connectable linked to the terminal. It should be solved by the modification suggested below. Add a unit test in any case.

Reply:
In IdentifiableExtensionXmlSerializerTest, some tests use TerminalmockXmlSerializer witch use this deprecated method